requests/exceptions.py

```diff
@@ -8,6 +8,16 @@
 """
 from urllib3.exceptions import HTTPError as BaseHTTPError
 
+try:
```

Comment:
We have logic for which simplejson/json exceptions we're using spread across a few places now. Is there a compelling reason not to simply import JSONDecodeError from compat since it's already been resolved and supplying it as a single parent class?

Reply:
@nateprewitt We want to know which properties the `JSONDecodeError` has, since it may have different properties depending on compatibility. That requires an extra check on import.

Reply:
@steveberdy I'm not sure I'm following, could you clarify which properties you're referring to? We have 3 parent classes that we'd ever use in our environment, `json.JSONDecodeError`, `simplejson.JSONDecodeError`, and `ValueError`. Only one of those can be relevant in a given execution of Requests.

What I was asking about was if we did:

```python
from .compat import JSONDecodeError as CompatJSONDecodeError

[...]

class JSONDecodeError(InvalidJSONError, CompatJSONDecodeError):
```

We've already resolved these values once in compat, so I wanted to know why we're doing it again here.

Reply:
For Python 3, in the `compat.py` file itself, reference to the `json` package directly, whether aliased as `json` or actually the `json` package itself, would always look for the `json` package rather than whatever was aliased as it. So, just in the `compat.py` file itself, I used the check. From all other scripts that have `from .compat import json`, however, it works as expected.

Reply:
Did you have any more questions about the code?

Reply:
Hey @steveberdy, I think there may still be some confusion. If you open the _Files changed_ tab, you should be able to see the full context of this thread. I've written out the [full diff](https://github.com/nateprewitt/requests/commit/665cf30861fb356d84103808e240d6940e8ff3d6) from the suggestion above to minimize ambiguity. You should find this passes the test suite without issue, so please let me know if there were other concerns you had.

While we're here it would also be good to add a test exercising this change. Something to the effect of the example below should be sufficient:

```
def test_json_decode_compatibility(self, httpbin):
    r = requests.get(httpbin('bytes/20'))
    with pytest.raises(requests.exceptions.JSONDecodeError):
        r.json()
```

This won't cover simplejson in our general test suite, but I've done a one off run to ensure that's working.

Reply:
Thank you, that makes total sense. I made those changes and I'm running workflows right now.